falcon/app_helpers.py

```diff
@@ -183,7 +184,8 @@ def default_serialize_error(req, resp, exception):
 
     if preferred is not None:
         if preferred == 'application/json':
-            resp.body = exception.to_json()
+            dumps = getattr(resp.options.media_handlers.get(MEDIA_JSON), 'dumps', None)
+            resp.body = exception.to_json(dumps)
         else:
             # NOTE(caselit): to_xml already returns bytes
             resp.data = exception.to_xml()
```

Comment:
Just a reminder to make sure the docs point out that this behavior only applies to JSON.